### PR TITLE
[5.2] Add Filesystem tests.

### PR DESCRIPTION
**Pull request by bancur**
Added 3 tests for previously untested Filesystems requireOnce, copy and isFile methods.

### CODE REVIEW

File: tests/Filesystem/FilesystemTest.php [hunk 1]
```diff
@@ -324,4 +324,40 @@ public function testSharedGet()
         $this->assertTrue($result === 1);
         @unlink(__DIR__.'/file.txt');
     }
+
+    public function testRequireOnceRequiresFileProperly()
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: This test will break if you run phpunit with multiple runs

Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: Or rather, it won't really test anything properly.

Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: You should call requireOnce more than one to check it only requires the file once.

File: tests/Filesystem/FilesystemTest.php [hunk 2]
```diff
@@ -324,4 +324,40 @@ public function testSharedGet()
...
+    }
+
+    public function testIsFileChecksFilesProperly()
+    {
+        $filesystem = new Filesystem;
+        mkdir(__DIR__.'/foo');
+        file_put_contents(__DIR__.'/foo/foo.txt', 'contents');
+        $this->assertTrue($filesystem->isFile(__DIR__.'/foo/foo.txt'));
+        $this->assertFalse($filesystem->isFile(__DIR__.'./foo'));
+        @unlink('/foo/foo.txt');
```
Comment (raphaelsaunier) [inline tests/Filesystem/FilesystemTest.php]: This line should be:
```php
@unlink(__DIR__ . '/foo/foo.txt');
```

Currently, this leaves the `/foo/foo.txt` in the tests folder and breaks several tests if the test suite is run a second time.

**Pull request merged**